packages/sdk/src/evm/core/classes/contract-wrapper.ts

```diff
@@ -510,6 +513,23 @@ export class ContractWrapper<
     // Parse the revert reason from the error
     const reason = parseRevertReason(error);
 
+    // Get contract sources for stack trace
+    let sources: ContractSource[] | undefined = undefined;
+    const storage = new ThirdwebStorage(); // Take thirdweb storage in constructor at some point?
+    try {
+      const metadata = await fetchContractMetadataFromAddress(
```

Comment:
Since we're fetching this everytime, how about putting the name of the contract in the error as well. That should be almost always available